### PR TITLE
#170764601 rate accommodation facility

### PR DESCRIPTION
**Pull request by niyodusengaclement**
#### What does this PR do?
Add rating center functionality
#### Description of Task to be completed?
-verify if the user is logged in
-verify if the user had stayed in the accommodation he wants to rate
-validate rating star(s) if is less or equal to five
-add rating star(s) and get a rating average of the accommodation
#### How should this be manually tested?
Clone this repo, cd into the project directory and run the following commands
 - npm install (install all dependencies)
 - npm run dev (start the server)
Open postman and go to localhost:<port>/api/accommodations/:accommodationId/rate
put the token in header and rating in body
#### What are the relevant pivotal tracker stories?
[#170764601](https://www.pivotaltracker.com/story/show/170764601)

### CODE REVIEW

File: src/middlewares/rating.middleware.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  static async RatingValidate(req, res, next) {
+    const { error } = RatingSchema.creatSchema(req.body);
+    try {
+      if (error) {
+        const response = new Response(res, 400, error.message);
+        return response.sendErrorMessage();
+      }
+      return next();
+    } catch (err) {
+      const response = new Response('Internal server error');
```
Comment (bihire) [inline src/middlewares/rating.middleware.js]: don' t you need to pass the response and status code in this instance call

File: src/middlewares/rating.middleware.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      where: {
+        user_id: userId,
+        accommodation_id: accommodationId
+      }
+    });
+
+    if (grantUser) {
+      return next();
+    }
+    return res.status(401).json({
```
Comment (bihire) [inline src/middlewares/rating.middleware.js]: could you consistently use premade class  for Responses

File: src/controllers/rating.controller.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        return res.status(201).json({
+          status: 201,
+          message: 'You have successfully updated your rate',
+          data: {
+            accommodationId: dataValues.accommodation_id,
+            ratingAvg: parseFloat(ratingAvg),
+          },
+        });
+      }
+      return res.status(500).json({
```
Comment (bihire) [inline src/controllers/rating.controller.js]: use premade class for response messages

**Review by ajimae — APPROVED** [review]
LGTM!

**Pull request merged**